dask/array/creation.py

```diff
@@ -64,13 +66,17 @@ def empty_like(a, dtype=None, chunks=None):
 
     a = asarray(a, name=False)
     return empty(
-        a.shape,
+        (shape or a.shape),
         dtype=(dtype or a.dtype),
-        chunks=(chunks if chunks is not None else a.chunks),
+        chunks=(
+            "auto"
+            if shape and (chunks is None)
```

Comment:
Why does the value of `shape` matter here?

Reply:
The main objective here is to make sure that when the user provides the outgoing shape and the chunks of `a` don't add up to outgoing shape, we can safely chunk the outgoing array appropriately. 



```python
In [2]: import dask.array as da

In [3]: a = da.arange(6).reshape(3, 2)

In [4]: a
Out[4]: dask.array<reshape, shape=(3, 2), dtype=int64, chunksize=(3, 2), chunktype=numpy.ndarray>

In [5]: da.ones_like(a, shape=(5,2))
Out[5]: dask.array<ones, shape=(5, 2), dtype=int64, chunksize=(5, 2), chunktype=numpy.ndarray>
```

Previously, this was giving:

```python
ValueError: Chunks do not add up to shape. Got chunks=((3,), (2,)), shape=(5, 2)
```